dlme_airflow/drivers/oai_xml.py

```diff
@@ -60,10 +63,26 @@ def _open_set(self):
                 )
             ]
         else:
-            oai_records = self._collection.ListRecords(
-                set=self.set, metadataPrefix=self.metadata_prefix, ignore_deleted=True
+            opts = {
+                "set": self.set,
+                "metadataPrefix": self.metadata_prefix,
+                "ignore_deleted": True,
+            }
+
+            if self.last_harvest_start_date and not self.full_harvest:
+                # need to use opts dict since "from" is a reserved word
+                opts["from"] = self.last_harvest_start_date.strftime("%Y-%m-%d")
+
+            logging.info(
+                "Harvesting %s with OAI ListRecords: %s", self.collection_url, opts
             )
 
+            try:
+                oai_records = self._collection.ListRecords(**opts)
+            except NoRecordsMatch:
+                logging.info("No new or updated records matching %s", opts)
+                oai_records = []
+
```

Comment:
Kinda weird that Sickle throws an exception when there are no records? I thought so, but maybe not...